ODM 0.3.1 2018-02-17: fix Georef pixelX_,pixelY_ from integer to doub…

yjmenezes opened the pull request:
please verify first 3 lines Georef.cpp for  formating log_.

Commit:
ODM 0.3.1 2018-02-17: fix Georef pixelX_,pixelY_ from integer to double and formating log_.

Commit:
fix pixelX_ to double, add idgcp_, log_ format

Commit:
fix gcp_list.txt

yjmenezes committed:
Update gcp_list.txt

pierotofy commented on modules/odm_georef/src/Georef.cpp:
```diff
@@ -25,7 +29,7 @@ GeorefGCP::~GeorefGCP()
 void GeorefGCP::extractGCP(std::istringstream &gcpStream)
 {
-    gcpStream >> x_ >> y_ >> z_ >> pixelX_ >> pixelY_ >> image_;
+    gcpStream >> x_ >> y_ >> z_ >> pixelX_ >> pixelY_ >> image_ >> idgcp_;
```
I think this would cause some issues if I use a GCP file which does not include IDs.

yjmenezes replied:
@pierotofy 
The idgcp_ can help on large datasets or can be used to label GCPs over Ortomosaic.
I upload a QGIS image.
We can comment and remove it to maintain compatibility with old gcp_list.txt layout but i think idgcp_ really can help.

pierotofy replied:
Indeed! I'm not saying we should remove it, I'm saying that we should support both formats. Some people might not need to label their GCPs.

yjmenezes replied:
@pierotofy ,

i think a try:{...}  inside
void GeorefGCP::extractGCP(std::istringstream &gcpStream)
could solve the idgcp_ issue keeping both layouts.
What do you think ?

pierotofy replied:
Only one way to find out. Could you compile the changes and test the odm_georef module with both varieties?

yjmenezes replied:
@pierotofy 

thx for your comments.
Ok, I will compile and test, this weekend.

pierotofy commented on modules/odm_georef/src/Georef.hpp:
```diff
@@ -46,10 +46,11 @@ struct GeorefGCP
 
-    int pixelX_;            /**< The pixels x-position for the GCP in the corresponding image **/
-    int pixelY_;            /**< The pixels y-position for the GCP in the corresponding image **/
+    double pixelX_;            /**< The pixels x-position for the GCP in the corresponding image **/
```
Good catch, since GCPs support floats we have been effectively truncating the precision here.

pierotofy reviewed (commented):
I think this is mostly good, could you please explain a bit about the purpose of adding IDs to each GCP?

We should also support older GCPs that don't use this new format (see comments).

yjmenezes committed:
Add files via upload

yjmenezes committed:
gcp_list.txt with # comments 

idgcp_ can help with large datasets or can be used to label GCP over orthomosaic under QGIS.

yjmenezes committed:
gcp_list.txt old and new, both working

yjmenezes committed:
gcp_list.txt old and new, both working

yjmenezes committed:
running ok with old and new gcp_list.txt layout.

yjmenezes committed:
Update odm_docker_readme.txt

Commit:
patch gcp_list.txt and External-OpenSfM.cmake

dakotabenjamin committed:
Merge branch 'master' into master
